pkg/reconciler/workload/resource/resource_reconcile.go

```diff
@@ -93,17 +117,16 @@ func (c *Controller) reconcileResource(ctx context.Context, lclusterName logical
 
 		syncTargetKey := strings.TrimPrefix(f, syncershared.SyncerFinalizerNamePrefix)
 		logger = logger.WithValues("syncTargetKey", syncTargetKey)
-		objs, err := c.syncTargetIndexer.ByIndex(indexers.SyncTargetsBySyncTargetKey, syncTargetKey)
+		syncTarget, err := c.getSyncTargetFromKey(syncTargetKey)
 		if err != nil {
-			logger.Error(err, "error getting SyncTarget via index")
-			continue
+			logger.Error(err, "error checking if sync target key exists")
+			return nil
```

Comment:
What is the likelihood of getting an error here?

Reply:
@jmprusi

Reply:
Thanks. There is no way a user would be able to know they hit this condition. @sttts @davidfestal do you have any concerns about this? It seems like it should "never" happen?

Reply:
I'm not very comfortable with returning nil here. Nothing would be patched at all.
I'd be OK with just cancelling the finalizers cleanup loop, and not patch the finalizers, and log an error, but still patch the labels and the annotations.

Reply:
Changed it to "continue" instead of return nil.